lib/determinator/retrieve/http_retriever.rb

```diff
@@ -13,10 +13,11 @@ def retrieve(name)
         begin
           response = @connection.get("/features/#{name}")
           return Determinator::Serializers::JSON.load(response.body) if response.status == 200
+          return nil if response.status == 404
```

Comment:
I wonder if there is a chance we model this better without having 3 different return types. Why not return always nil when it's non-200 and never cache that result?

Reply:
>Why not return always nil when it's non-200 and never cache that result?

That's what the option in the first half of the PR allows. Agree this need a remodelling with a proper 404Type.